### PR TITLE
Iox #453 add c version to iceperf example

### LINKED ISSUE #453: Extend iceperf example to C API
## Brief feature description

We should extend the ice perf example with an laurel and hardy which are using the C API. This API doesn't not use constructs like `optional`, `expected` and `sample` so that we may see the performance costs of those abstractions.

## Detailed information

Additionally, it could be a target value of future optimization of the `optional`, `expected` and the new C++ API

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Closes #453

### CODE REVIEW

**Review by elBoberido — APPROVED** [review]

**Review by elBoberido — DISMISSED** [review]

File: iceoryx_examples/iceperf/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +23,7 @@ if ( NOT ICEORYX_CXX_STANDARD )
     include(IceoryxPlatform)
 endif ( NOT ICEORYX_CXX_STANDARD )
 
-add_executable(iceperf-laurel iceperf_laurel.cpp base.cpp iceoryx.cpp uds.cpp mq.cpp)
+add_executable(iceperf-laurel iceperf_laurel.cpp base.cpp example_common.cpp iceoryx.cpp iceoryx_c.cpp uds.cpp mq.cpp)
```
Comment (dkroenke) [inline iceoryx_examples/iceperf/CMakeLists.txt:26]: please add `find_package(iceoryx_binding_c)` above

**Review by dkroenke — CHANGES_REQUESTED** [review]

File: iceoryx_examples/iceperf/CMakeLists.txt (hunk 1 above)
Comment (elfenpiff) [reply iceoryx_examples/iceperf/CMakeLists.txt:27]: thanks, already done!

**Review by elBoberido — DISMISSED** [review]

**Review by dkroenke — APPROVED** [review]

File: iceoryx_examples/iceperf/iceperf_laurel.cpp [hunk 2]
```diff
@@ -71,30 +75,49 @@ void leaderDo(IcePerfBase& ipcTechnology, int64_t numRoundtrips)
 
 int main(int argc, char* argv[])
 {
+    Benchmarks benchmark = Benchmarks::ALL;
```
Comment (budrus) [inline iceoryx_examples/iceperf/iceperf_laurel.cpp:78]: I guess you'll have a problem if laurel and hardy are not started with the same benchmark option...

File: iceoryx_examples/iceperf/CMakeLists.txt [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 # Build iceperf example
 cmake_minimum_required(VERSION 3.5)
 project(example_iceperf)
 
 find_package(iceoryx_posh CONFIG REQUIRED)
+find_package(iceoryx_binding_c CONFIG REQUIRED)
```
Comment (budrus) [inline iceoryx_examples/iceperf/CMakeLists.txt:20]: Is it then possible to build the examples without an activated C binding?

File: iceoryx_examples/iceperf/iceperf_laurel.cpp (hunk 2 above)
Comment (elfenpiff) [reply iceoryx_examples/iceperf/iceperf_laurel.cpp:78]: Yes. But you are able to mix the C++ and C API.

File: iceoryx_examples/iceperf/CMakeLists.txt (hunk 3 above)
Comment (elfenpiff) [reply iceoryx_examples/iceperf/CMakeLists.txt:20]: No.

**Review by dkroenke — APPROVED** [review]

File: iceoryx_examples/iceperf/CMakeLists.txt (hunk 3 above)
Comment (budrus) [reply iceoryx_examples/iceperf/CMakeLists.txt:20]: will be addressed in #452

**Review by budrus — APPROVED** [review]

**Pull request merged**